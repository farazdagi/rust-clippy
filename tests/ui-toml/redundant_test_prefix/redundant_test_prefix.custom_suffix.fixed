//@revisions: default outside_cfg_test custom_suffix
//@[default] rustc-env:CLIPPY_CONF_DIR=tests/ui-toml/redundant_test_prefix/default
//@[outside_cfg_test] rustc-env:CLIPPY_CONF_DIR=tests/ui-toml/redundant_test_prefix/outside_cfg_test
//@[custom_suffix] rustc-env:CLIPPY_CONF_DIR=tests/ui-toml/redundant_test_prefix/custom_suffix
//@compile-flags: --test
#![allow(dead_code)]
#![warn(clippy::redundant_test_prefix)]

fn main() {}

mod tests_no_annotations {
    use super::*;

    #[test]
    fn has_annotation() {
        //~[outside_cfg_test]^ redundant_test_prefix
    }

    fn no_annotation() {}
}

#[test]
fn main_module_has_annotation() {
    //~[outside_cfg_test]^ redundant_test_prefix
}

fn test_main_module_no_annotation() {}

fn foo() {}

#[cfg(test)]
#[test]
fn foo_ok() {
    //~^ ERROR: redundant `test_` prefix in test function name
    //~| HELP: consider removing the `test_` prefix (suffix avoids name conflict)

    todo!()
    // Has prefix, has `#[test]` attribute, within a `#[cfg(test)]`.
    // Collision with existing function, so suffix is added.
}

fn bar() {}

#[test]
fn bar_ok() {
    //~[custom_suffix]^ redundant_test_prefix

    todo!()
    // Has prefix, has `#[test]` attribute.
    // NOT within a `#[cfg(test)]`, but the lint is enabled for integration tests.
    // Collision with existing function, so suffix is added.
}

#[cfg(test)]
mod tests {}
