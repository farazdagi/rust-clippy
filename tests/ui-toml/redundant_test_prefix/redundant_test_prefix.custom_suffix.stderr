error: redundant `test_` prefix in test function name
  --> tests/ui-toml/redundant_test_prefix/redundant_test_prefix.rs:15:8
   |
LL |     fn test_has_annotation() {
   |        ^^^^^^^^^^^^^^^^^^^ help: consider removing the `test_` prefix: `has_annotation`
   |
   = note: `-D clippy::redundant-test-prefix` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::redundant_test_prefix)]`

error: redundant `test_` prefix in test function name
  --> tests/ui-toml/redundant_test_prefix/redundant_test_prefix.rs:23:4
   |
LL | fn test_main_module_has_annotation() {
   |    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider removing the `test_` prefix: `main_module_has_annotation`

error: redundant `test_` prefix in test function name
  --> tests/ui-toml/redundant_test_prefix/redundant_test_prefix.rs:33:4
   |
LL | fn test_foo() {
   |    ^^^^^^^^ help: consider removing the `test_` prefix (suffix avoids name conflict): `foo_ok`

error: redundant `test_` prefix in test function name
  --> tests/ui-toml/redundant_test_prefix/redundant_test_prefix.rs:45:4
   |
LL | fn test_bar() {
   |    ^^^^^^^^ help: consider removing the `test_` prefix (suffix avoids name conflict): `bar_ok`

error: aborting due to 4 previous errors

